Styles/Classes/TextStyle.swift

```diff
@@ -86,25 +89,43 @@ public final class TextStyle: NSObject {
         self.effects = effects
     }
 
+    public static let empty = TextStyle(attributes: [:], effects: [])
+
     @objc public func apply(to text: String) -> NSAttributedString {
         let result = NSMutableAttributedString(string: text, attributes: attributes)
         for effect in effects {
-            let effectAttributes = effect.attributes
-            for range in effect.ranges(in: text) {
-                result.addAttributes(effectAttributes, range: range)
-            }
+            effect.apply(to: result, baseAttributes: attributes)
         }
         return result
     }
 
-    public func updating(_ properties: Property ...) -> TextStyle {
+    // MARK:- Combining Styles
+
+    public func updating(_ properties: Property...) -> TextStyle {
         let newAttributes = attributes.merging(properties.attributes, uniquingKeysWith: { $1 })
         return TextStyle(attributes: newAttributes, effects: effects)
     }
 
     public static func +(left: TextStyle, right: TextStyle) -> TextStyle {
         let newAttributes = left.attributes.merging(right.attributes, uniquingKeysWith: { $1 })
-        let newEffects = left.effects.updating(right.effects)
+        let newEffects = left.effects + right.effects
         return TextStyle(attributes: newAttributes, effects: newEffects)
     }
+
+    // MARK:- Updating effects
+    public func appending(_ other: TextEffect...) -> TextStyle {
```

Comment:
Does it filter same TextEffect if exists?

Reply:
not right now. Will do.